src/nnsight/__init__.py

```diff
@@ -548,3 +549,88 @@ def set_module_tensor_to_device(
 apply = GlobalTracingContext.GLOBAL_TRACING_CONTEXT.apply
 log = GlobalTracingContext.GLOBAL_TRACING_CONTEXT.log
 cond = GlobalTracingContext.GLOBAL_TRACING_CONTEXT.cond
+
+import inspect
+
+from . import util
+from .intervention import InterventionProxy
+
+
+def trace(fn: Callable):
+    """Helper decorator to add a function to the intervention graph via `.apply(...)`.
+    This is opposed to entering the function during tracing and tracing all inner operations.
+
+    Args:
+        fn (Callable): Function to apply.
+
+    Returns:
+        Callable: Traceable function.
+    """
+
+    @wraps(fn)
+    def inner(*args, **kwargs):
+
+        return apply(fn, *args, **kwargs)
+
+    return inner
```

Comment:
I see this pattern defined in numerous locations (`__init__.py`, `GraphBasedContext.py`). Do you think this could be reused as a `util` function (with a descriptive docstring)?